assets/blocks/jobs/types.jsx

```diff
@@ -0,0 +1,243 @@
+
+/**
+ * WordPress Dependencies.
+ */
+const { Component } = wp.element,
+	{ __ } = wp.i18n,
+	{ addFilter } = wp.hooks,
+	{
+		withAPIData,
+		CheckboxControl,
+		ToggleControl,
+	} = wp.components;
+
+/**
+ * External Dependencies.
+ */
+import _ from 'lodash';
+
+
+/**
+ * Job Types block attributes.
+ */
+const jobTypesAttributesConfig = {
+	showJobTypeFilters: {
+		type: 'boolean',
+		default: true,
+	},
+	includedJobTypes: {
+		type: 'string', // JSON object
+		default: '{}',
+	},
+	allJobTypes: {
+		type: 'string', // JSON array
+		default: '[]',
+	}
+};
+addFilter(
+	'wpjm_block_jobs_attributes_config',
+	'addJobTypesAttributes',
+	( attributesConfig ) => (
+		Object.assign( attributesConfig, jobTypesAttributesConfig )
+	)
+);
+
+
+/**
+ * Transform a job type attribute value from a JSON-encoded object to a
+ * comma-separated string.
+ *
+ * @param {string} value The JSON-encoded job type attribute value.
+ *
+ * @return {?string} Comma-separated value string for the attribute, or null.
+ */
+function transformJobTypeValue( value ) {
+	let obj = {};
+	try {
+		obj = JSON.parse( value );
+	} catch ( error ) {
+		// On error, just use the empty object.
+	}
+
+	// Get the keys whose values are truthy.
+	let keysToInclude = _.pickBy( obj, ( val ) => val );
+
+	// Create a comma-separated string.
+	value = Object.keys( keysToInclude ).join( ',' );
+
+	if ( ! value ) {
+		value = null;
+	}
+
+	return value;
+}
+
+/**
+ * Transform the Job Types attributes into shortcode parameters.
+ *
+ * @param {Object} shortcodeParams The incoming shortcode parameters.
+ * @param {Object} attributes      The block attributes.
+ *
+ * @return {Object} The new shortcode parameters.
+ */
+function getShortcodeParameters( shortcodeParams, attributes ) {
+	let jobTypes = transformJobTypeValue( attributes.includedJobTypes );
+
+	if ( attributes.showJobTypeFilters ) {
+		if ( jobTypes ) {
+			shortcodeParams.selected_job_types = jobTypes;
```

Comment:
Same re-assign issue as above. The problem here is that this will re-assign the object properties in the original scope, aka, side-effects!!

Reply:
Done.